metadata-io/src/main/java/com/linkedin/metadata/entity/DeleteEntityService.java

```diff
@@ -0,0 +1,254 @@
+      gmce.setEntityUrn(relatedUrn);
+      gmce.setChangeType(ChangeType.UPSERT);
+      gmce.setEntityType(relatedEntityName);
+      gmce.setAspectName(aspectName);
+      gmce.setAspect(GenericRecordUtils.serializeAspect(updatedAspect));
+
+      final AuditStamp auditStamp = new AuditStamp().setActor(UrnUtils.getUrn(Constants.SYSTEM_ACTOR)).setTime(System.currentTimeMillis());
+      final EntityService.IngestProposalResult ingestProposalResult = _entityService.ingestProposal(gmce, auditStamp);
+
+      if (!ingestProposalResult.isDidUpdate()) {
+        log.warn("Aspect update did not update metadata graph. Before {}, after: {}", aspect, updatedAspect);
+      }
+    }
+  }
+
+  /**
+   * Utility method that attempts to find Aspect information as well as the associated path spec for a given urn that
+   * has a relationship of type `relationType` to another urn
+   * @param urn               The urn of the entity for which we want to find the aspect that relates the urn to
+   *                          `relatedUrn`.
+   * @param relatedUrn        The urn of the related entity in which we want to find the aspect that has a relationship
+   *                          to `urn`.
+   * @param relationType      The relationship type that details how the `urn` entity is related to `relatedUrn` entity.
+   * @param relatedEntitySpec The entity spec of the related entity.
+   * @return An {@link java.util.Optional} object containing the aspect content & respective spec that contains the
+   * relationship between `urn` & `relatedUrn`.
+   */
+  public java.util.Optional<Pair<EnvelopedAspect, AspectSpec>> getAspectReferringTo(final Urn urn, final Urn relatedUrn,
+      final String relationType, final @Nonnull EntitySpec relatedEntitySpec) {
+
+    // Find which aspects are the candidates for the relationship we are looking for
+    final Map<String, AspectSpec> aspectSpecs = relatedEntitySpec.getAspectSpecMap()
+        .entrySet()
+        .stream()
+        .filter(aspectSpec -> !aspectSpec.getValue().findRelationshipFor(relationType, urn.getEntityType()).isEmpty())
+        .collect(Collectors.toMap(Map.Entry::getKey, Map.Entry::getValue));
+
+    // FIXME: Can we not depend on entity service?
+    final EntityResponse entityResponse;
+    try {
+      entityResponse = _entityService.getEntityV2(relatedUrn.getEntityType(),
+          relatedUrn,
+          aspectSpecs.keySet());
+    } catch (URISyntaxException e) {
+      log.error("Unable to retrieve entity data for relatedUrn " + relatedUrn, e);
+      return java.util.Optional.empty();
+    }
+
+    // Find aspect which contains the relationship with the value we are looking for
+    return entityResponse
+        .getAspects().values()
+        .stream()
+        // Get aspects which contain the relationship field specs found above
+        .filter(aspect -> aspectWithRelationshipTo(urn.getEntityType(), relationType, aspectSpecs, aspect))
```

Comment:
Don't we already know it does from the above?

Reply:
So the part before was to extract the aspect specs, this part is to extract the aspect payload (record template)